rotkehlchen/exchanges/bitstamp.py

```diff
@@ -0,0 +1,711 @@
+        endpoint: Literal['user_transactions']
+        response_case: Literal['trades', 'asset_movements']
+        if case == 'trades':
+            endpoint = 'user_transactions'
+            raw_result_type_filter = USER_TRANSACTION_TRADE_TYPE
+            response_case = 'trades'  # same than `case`
+            case_pretty = 'trade'
+        elif case == 'asset_movements':
+            endpoint = 'user_transactions'
+            raw_result_type_filter = USER_TRANSACTION_ASSET_MOVEMENT_TYPE
+            response_case = 'asset_movements'  # same than `case`
+            case_pretty = 'asset movement'
+        else:
+            raise AssertionError(f'Unexpected Bitstamp case: {case}.')
+
+        # Check required options per endpoint
+        if endpoint == 'user_transactions':
+            assert options['limit'] == API_MAX_LIMIT
+            assert options['since_id'] >= USER_TRANSACTION_MIN_SINCE_ID
+            assert options['sort'] == USER_TRANSACTION_SORTING_MODE
+            assert options['offset'] == 0
+
+        call_options = options.copy()
+        limit = options.get('limit', API_MAX_LIMIT)
+        results = []
+        while True:
+            response = self._api_query(
+                endpoint=endpoint,
+                method='post',
+                options=call_options,
+            )
+            if response.status_code != HTTPStatus.OK:
+                return self._process_unsuccessful_response(
+                    response=response,
+                    case=response_case,
+                )
+            try:
+                response_list = rlk_jsonloads_list(response.text)
+            except JSONDecodeError:
+                msg = f'Bitstamp returned invalid JSON response: {response.text}.'
+                log.error(msg)
+                self.msg_aggregator.add_error(
+                    f'Got remote error while querying Bistamp trades: {msg}',
+                )
+                no_results: Union[List[Trade], List[AssetMovement]] = []  # type: ignore
+                return no_results
+
+            has_results = False
+            is_result_timesamp_gt_end_ts = False
+            for raw_result in response_list:
+                if (
+                    # NB: skipping by type is just for user_transactions, ignoring type error
+                    endpoint == 'user_transactions' and  # type: ignore
+                    raw_result['type'] not in raw_result_type_filter
+                ):
+                    continue
+                try:
+                    result_timestamp = iso8601ts_to_timestamp(raw_result['datetime'])
+
+                    if result_timestamp > end_ts:
+                        is_result_timesamp_gt_end_ts = True  # prevent extra request
+                        break
+
+                    if case == 'trades':
+                        result = self._deserialize_trade(raw_result)
+                    elif case == 'asset_movements':
+                        # TODO: fix this type...
```

Comment:
I have a type error here. I've tried to declare `result: Union[Trade, AssetMovement]` above but nothing.